sw/device/silicon_creator/lib/crc32.c

```diff
@@ -34,13 +43,23 @@ void crc32_add8(uint32_t *ctx, uint8_t byte) {
       *ctx ^= kCrc32Poly;
     }
   }
+#endif
 }
 
 void crc32_add32(uint32_t *ctx, uint32_t word) {
+#ifdef OT_PLATFORM_RV32
+  uint32_t state = *ctx ^ word;
```

Comment:
I suspect we can get rid of these repetitive memory reads and writes and just keep the context in a register.

Reply:
Compiling with `-O2` appears to do this.